lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Balance.hs

```diff
@@ -510,15 +515,6 @@ performSelection minCoinFor costFor bundleSizeAssessor criteria
                 , balanceRequired
                 }
   where
-    SelectionCriteria
-        { outputsToCover
-        , utxoAvailable
-        , selectionLimit
-        , extraCoinSource
-        , assetsToMint
-        , assetsToBurn
-        } = criteria
```

Comment:
Hi @rvl!

I wonder if we can avoid using `RecordWildCards` here?

My personal experience is that with this extension, I find it much harder to determine where symbols are introduced and consumed, especially in long functions such as this one. I suspect I'm not alone in this! With `NamedFieldPuns`, it's a simple matter to find out where things are introduced and consumed, even if it is a bit annoying to have to list out the fields you want.

Haskell Language Server might make this easier, but that's not available in all contexts (especially in a code review context).